src/fastapibootcamp/handlers/external.py

```diff
@@ -317,3 +330,135 @@ async def post_log_demo(
         greeting=greeting_templates[data.language].format(name=data.name),
         language=data.language,
     )
+
+
+# =============================================================================
+# Lesson 5: Handling internal errors
+#
+# Sometimes things go wrong in your application. A database doesn't respond,
+# an external service is down, or some bug in your code causes an exception.
+# By default, FastAPI returns a 500 Internal Server Error response when an
+# uncaught exception occurs. Safir provides the ability to report these
+# uncaught errors to a Slack channel through an incoming webhook.
+#
+# To hook up this Slack reporting, you need to take the following steps
+# 1. Add a slack_webhook_url field to the configuration (config.py)
+# 2. In main.py, configure the SlackRouteErrorHandler middleware
```

Comment:
Technically, there's no middleware here, just a custom route handler. That acts a lot like middleware, but it's not actually Starlette middleware.